test_email_processor.py

```
@@ -0,0 +1,135 @@
+        ]
+
+        for name, from_header, is_blocked, is_allowed, cat1, cat2, cat_blocked, expected in test_cases:
+            with self.subTest(name=name):
+                # Reset mocks
+                self.fetcher.reset_mock()
+                mock_cat1.reset_mock()
+                mock_cat2.reset_mock()
+                
+                # Setup message
+                self.msg.get.side_effect = lambda x, default='': {
+                    'From': from_header,
```

Ruff (0.8.0)

Function definition does not bind loop variable `from_header`

(B023)